src/simulation.hpp

```diff
@@ -703,6 +708,14 @@ template <typename problem_t> void AMRSimulation<problem_t>::evolve()
 			WritePlotFile();
 		}
 
+		// Writing Plot files at time intervals
+		double next_plot_file_time = next_plot_file_num * plotTimeInterval_;
+		if (plotTimeInterval_ > 0 && ((next_plot_file_time > cur_time) && (next_plot_file_time < (cur_time + dt_[0])))) {
+			next_plot_file_num += 1;
```

Comment:
I think this might be clearer if you change it to `next_plot_file_time += plotTimeInterval_`. Then `next_plot_file_num` is not needed.

Reply:
I can do this.